packages/badge-service/badgeGenerator/badgeGeneratorSpec.ts

```diff
@@ -0,0 +1,193 @@
+            delete context.bindingData.branch;
+
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(200);
+            expect(httpHelpers.getContent).calledWith('https://img.shields.io/badge/mutation%20score-79-orange.svg');
+        }),
+        it('Should call shields.io with the correct parameters', async () => {
+            // Arrange
+            sandbox.stub(httpHelpers, 'getContent').resolves('your image');
+
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(200);
+            expect(httpHelpers.getContent).calledWith('https://img.shields.io/badge/mutation%20score-97.8-green.svg');
+        }),
+        it('Should return correct cache headers', async () => {
+            // Arrange
+            sandbox.stub(httpHelpers, 'getContent').resolves('your image');
+
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(200);
+            expect(context.res.headers).to.deep.equal({
+                'Cache-Control': 'no-cache, no-store, must-revalidate',
+                'Pragma': 'no-cache'
+            });
+        }),
+        it('Should return a green shield with a score >= 80', async () => {
+            // Arrange
+            sandbox.stub(httpHelpers, 'getContent').resolves('green');
+            mutationScore.score= 80;
+            
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(200);
+            expect(httpHelpers.getContent).calledWith('https://img.shields.io/badge/mutation%20score-80-green.svg');
+        }),
+        it('Should return an orange shield with a 60 <= score < 80', async () => {
+            // Arrange
+            sandbox.stub(httpHelpers, 'getContent').resolves('orange');
+            mutationScore.score= 79.9;
+            
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(200);
+            expect(httpHelpers.getContent).calledWith('https://img.shields.io/badge/mutation%20score-79.9-orange.svg');
+        }),
+        it('Should return an orange shield with a score = 60', async () => {
+            // Arrange
+            sandbox.stub(httpHelpers, 'getContent').resolves('orange');
+            mutationScore.score= 60;
+            
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(200);
+            expect(httpHelpers.getContent).calledWith('https://img.shields.io/badge/mutation%20score-60-orange.svg');
+        }),
+        it('Should return a red shield with a score < 60', async () => {
+            // Arrange
+            sandbox.stub(httpHelpers, 'getContent').resolves('red');
+            mutationScore.score= 59.9;
```

Comment:
I'm still missing a scenario where mutation score is a not-rounded number. Like `23.1412321241231431`. I would hope that the mutation score is rounded to `23.1`

Reply:
Not currently supported, do we even want to accept/store those scores? i.e. we don't accept it when posting the score, or we round it then?

Reply:
we round it at get time, sounds better to me, keep source data integrity in check. I've updated the tests